### PR TITLE
link/Coff: simplify file structure by collapsing all files into Coff.zig

### PR DESCRIPTION
**Pull request by kubkon**

### CODE REVIEW

**Review by andrewrk — APPROVED** [review]
This is no functional changes right?

Can we wait until I get  #21700 landed? Doing my best to get it merge-ready today.

**Comment by kubkon** [comment]
> This is no functional changes right?

Correct.

> 
> Can we wait until I get #21700 landed? Doing my best to get it merge-ready today.

Yep, I'm not blocked by this so happy to wait :+1:

**Pull request merged**
